imap_processing/spice/geometry.py

```diff
@@ -443,3 +444,74 @@ def instrument_pointing(
     if isinstance(et, typing.Collection):
         return np.rad2deg([spice.reclat(vec)[1:] for vec in pointing])
     return np.rad2deg(spice.reclat(pointing)[1:])
+
+
+def cartesian_to_spherical(
+    v: NDArray,
+) -> NDArray:
+    """
+    Convert cartesian coordinates to spherical coordinates.
+
+    Parameters
+    ----------
+    v : np.ndarray
+        A NumPy array with shape (n, 3) where each
+        row represents a vector
+        with x, y, z-components.
+
+    Returns
+    -------
+    spherical_coords : np.ndarray
+        Spherical coordinate.
+    """
+    vx = v[:, 0]
+    vy = v[:, 1]
+    vz = v[:, 2]
+
+    # Magnitude of the velocity vector
+    magnitude_v = np.sqrt(vx**2 + vy**2 + vz**2)
+
+    vhat_x = -vx / magnitude_v
+    vhat_y = -vy / magnitude_v
+    vhat_z = -vz / magnitude_v
+
+    # Elevation angle (angle from the z-axis, range: [-pi/2, pi/2])
+    el = np.arcsin(vhat_z)
+
+    # Azimuth angle (angle in the xy-plane, range: [0, 2*pi])
+    az = np.arctan2(vhat_y, vhat_x)
```

Comment:
I'm wondering if some of this can be streamlined by:
```suggestion
    magnitude_v = np.sqrt(np.sum(v**2, axis=-1))
    vhat = -v / magnitude_v
    # Elevation angle (angle from the z-axis, range: [-pi/2, pi/2])
    el = np.arcsin(vhat[..., 2])
    # Azimuth angle (angle in the xy-plane, range: [0, 2*pi])
    az = np.arctan2(vhat[..., 1], vhat[..., 0])
```